service/dap/server.go

```diff
@@ -1591,6 +1613,17 @@ func (s *Server) childrenToDAPVariables(v *fullyQualifiedVariable) ([]dap.Variab
 				Value:              cvalue,
 				VariablesReference: cvarref,
 			}
+
+			if c.Kind == reflect.Array || c.Kind == reflect.Slice || c.Kind == reflect.Map {
+				children[i].IndexedVariables = int(c.Len)
+			}
+
+			// TODO(suzmue): Paging named variables are not yet supported by the VS Code UI
```

Comment:
I was actually thinking about this some more and I think it will make sense to use named variables to include things like "len", "cap", "string value" or whatever else and keep indexed for map values and arrays.

I'll remove this comment since it now seems like indexed is the right choice for maps.